client/src/components/Login.jsx

```diff
@@ -1,49 +1,106 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
+import { Redirect } from 'react-router-dom';
 
 class Login extends React.Component {
   constructor (props) {
     super(props);
 
     this.state = {
       hostName: '',
-      zipCode: ''
+      zipCode: '',
+      hostNameError: false,
+      zipCodeError: false,
+      formIsValid: false
     };
 
     this.handleChange = this.handleChange.bind(this);
+    this.checkFormCompletion = this.checkFormCompletion.bind(this);
     this.saveHost = this.saveHost.bind(this);
   }
 
+  componentDidMount () {
+    const options = {
+      enableHighAccuracy: false,
+      timeout: 10000,
+      maximumAge: 0
+    };
+
+    const success = (pos) => {
+      const hostGeo = {
+        lat: pos.coords.latitude,
+        lon: pos.coords.longitude
+      };
+      this.props.setTopLevelState('hostGeo', hostGeo);
+    };
+
+    const error = (err) => {
+      console.warn(`ERROR(${err.code}): ${err.message}`);
+    };
+
+    navigator.geolocation.getCurrentPosition(success, error, options);
+  }
+
   handleChange (e) {
     const value = e.target.value;
     const name = e.target.name;
     this.setState({ [name]: value });
   }
 
+  checkFormCompletion () {
+    if (!this.state.hostName) {
+      this.setState({ hostNameError: true });
+    } else {
+      this.setState({ hostNameError: false });
+    }
+
+    if (!this.props.hostGeo && !this.state.zipCode) {
+      this.setState({ zipCodeError: true });
+    } else {
+      this.setState({ zipCodeError: false });
+    }
+
+    if (this.state.hostName && (this.props.hostGeo || this.state.zipCode)) {
+      this.setState({ hostNameError: false });
+      this.setState({ zipCodeError: false });
+      this.saveHost();
+    }
+  }
+
   saveHost () {
     const hostDetails = {
       id: 0,
       guestName: this.state.hostName,
       order: []
     };
+
+    if (this.state.zipCode) {
+      this.props.setTopLevelState('hostZipCode', this.state.zipCode);
+    }
+
     this.props.setTopLevelState('guests', [hostDetails]);
+    this.setState({ formIsValid: true });
   }
 
   render () {
+    if (this.state.formIsValid) {
+      return <Redirect to='/find-restaurant' />;
+    }
+
     return (
       <div>
         <h2>Login</h2>
         <form>
           <label htmlFor="hostName">Host Name:</label>
           <input type="text" inputMode="text" name="hostName" value={this.state.hostName} onChange={this.handleChange} />
+          {this.state.hostNameError && <p className="error">Host Name is required.</p>}
```

Comment:
A easier way to check the input field is: if input is mandatory, you can add "required" at end
<input type="text" inputMode="text" name="hostName" value={this.state.hostName} onChange={this.handleChange} required />

Reply:
Ah got it thank you! Maybe we can update all our controlled inputs that require values to follow that pattern, since currently I don't think they are. Could be something I can take on in a separate PR.